lib/jellyfish/room.ex

```diff
@@ -278,8 +278,16 @@ defmodule Jellyfish.Room do
 
   @impl true
   def handle_call(:get_num_forwarded_tracks, _from, state) do
-    forwarded_tracks = Engine.get_num_forwarded_tracks(state.engine_pid)
-    {:reply, forwarded_tracks, state}
+    pid = state.engine_pid
+
+    try do
+      forwarded_tracks = Engine.get_num_forwarded_tracks(pid)
+      {:reply, forwarded_tracks, state}
+    catch
+      :exit, {:noproc, {GenServer, :call, [^pid, :get_num_forwarded_tracks, _timeout]}} ->
+        Logger.warning("Call to room #{state.id} failed")
+        {:reply, 0, state}
+    end
```

Comment:
I wonder how we should handle this. We call `get_num_forwarded_tracks` in RoomService in Task.async. If Engine crashesh, our room should exit anyway so the question is: should we crash room we engine crashesh or receive `DOWN` message from the Engine and close room in a more controlled way?

Reply:
![](https://emojis.slackmojis.com/emojis/images/1643514070/263/plusone.png?1643514070)

Reply:
I think we should stop the room process when we receive a `DOWN` message, but here we must also catch an error to not fail a `Task.async`.